src/openfermion/utils/_low_rank_test.py

```diff
@@ -190,3 +191,56 @@ def test_rank_reduction(self):
                 decomposed_operator - fermion_hamiltonian)
             errors += [difference.induced_norm()]
         self.assertTrue(errors[3] <= errors[2] <= errors[1] <= errors[0])
+
+    def test_one_body_square_decomposition(self):
+
+        # Initialize a random two-body FermionOperator.
+        n_qubits = 4
+        random_operator = get_fermion_operator(
+            random_interaction_operator(n_qubits))
+
+        # Convert to chemist tensor.
+        constant, one_body_coefficients, chemist_tensor = \
+            get_chemist_two_body_coefficients(random_operator)
+
+        # Perform decomposition.
+        eigenvalues, one_body_squares, trunc_error = \
+            low_rank_two_body_decomposition(chemist_tensor)
+
+        # Build back two-body component.
+        for l in range(n_qubits ** 2):
+
+            # Get the squared one-body operator.
+            one_body_operator = FermionOperator()
+            for p, q in itertools.product(range(n_qubits), repeat=2):
+                term = ((p, 1), (q, 0))
+                coefficient = one_body_squares[l, p, q]
+                one_body_operator += FermionOperator(term, coefficient)
+            one_body_squared = one_body_operator ** 2
+
+            # Get the squared one-body operator via one-body decomposition.
+            density_density_matrix, basis_transformation_matrix = \
+                prepare_one_body_squared_evolution(one_body_squares[l])
+            two_body_operator = FermionOperator()
+            for p, q in itertools.product(range(n_qubits), repeat=2):
+                term = ((p, 1), (p, 0), (q, 1), (q, 0))
+                coefficient = density_density_matrix[p, q]
+                two_body_operator += FermionOperator(term, coefficient)
+
+            # Confirm that the rotations diagonalize the one-body squares.
+            hopefully_diagonal = basis_transformation_matrix.dot(
+                numpy.dot(one_body_squares[l],
+                          numpy.transpose(numpy.conjugate(
+                              basis_transformation_matrix))))
+            diagonal = numpy.diag(hopefully_diagonal)
+            difference = hopefully_diagonal - numpy.diag(diagonal)
+            self.assertAlmostEqual(0., numpy.amax(numpy.absolute(difference)))
+            density_density_alternative = numpy.outer(diagonal, diagonal)
+            difference = density_density_alternative - density_density_matrix
+            self.assertAlmostEqual(0., numpy.amax(numpy.absolute(difference)))
+
+            # Test spectra.
+            one_body_squared_spectrum = eigenspectrum(one_body_squared)
+            two_body_spectrum = eigenspectrum(two_body_operator)
+            difference = two_body_spectrum - one_body_squared_spectrum
+            self.assertAlmostEqual(0., numpy.amax(numpy.absolute(difference)))
```

Comment:
If we expect the resulting normal ordered operators to be the same also, maybe we can test that as well? Or is that too onerous?